actionpack/lib/action_dispatch/testing/integration.rb

```diff
@@ -50,11 +50,18 @@ def head(path, *args)
 
       # Follow a single redirect response. If the last response was not a
       # redirect, an exception will be raised. Otherwise, the redirect is
-      # performed on the location header. Any arguments are passed to the
-      # underlying call to `get`.
+      # performed on the location header. If the redirection is a 307 redirect,
+      # the same HTTP verb will be used when redirecting, otherwise a GET request
+      # will be performed. Any arguments are passed to the
+      # underlying request.
       def follow_redirect!(**args)
         raise "not a redirect! #{status} #{status_message}" unless redirect?
-        get(response.location, **args)
+
+        if response.status == 307
+          public_send(request.method.downcase, response.location, **args)
```

Comment:
Is `request.method` guaranteed to be a valid verb or are we `public_send`ing something potentially dangerous? This is also a test helper so it might not be important.

Reply:
After reading `ActionDispatch::Request`, I think this is fine 👍